src/ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.cpp

```diff
@@ -0,0 +1,61 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace LinearSolver::Schwarz {
+
+size_t overlap_extent(const size_t volume_extent,
+                      const size_t max_overlap) noexcept {
+  return std::min(max_overlap, volume_extent - 1);
+}
+
+template <size_t Dim>
+Index<Dim> overlap_extents(Index<Dim> extents, const size_t max_overlap,
+                           const size_t dimension) noexcept {
+  ASSERT(dimension < Dim, "Dimension '" << dimension << "' invalid in "
+                                        << "D");
+  extents[dimension] = overlap_extent(extents[dimension], max_overlap);
+  return extents;
+}
+
+double overlap_width(const size_t overlap_extent,
+                     const DataVector& collocation_points) noexcept {
+  ASSERT(overlap_extent < collocation_points.size(),
+         "Overlap extent is "
+             << overlap_extent
+             << " but must be strictly smaller than the number of grid points ("
+             << collocation_points.size() << ") to compute an overlap width.");
+  for (size_t i = 0; i < collocation_points.size() / 2; ++i) {
+    ASSERT(collocation_points[i] +
+                   collocation_points[collocation_points.size() - i - 1] <
+               1.e-14,
```

Comment:
Use `equal_within_roundoff`.  (The current version will pass if the error is negative.)